Tree.h

```diff
@@ -132,13 +132,21 @@ class LeafNode : public TreeNode<T> {
 
 // load a regression tree from Json
 template <class T>
-TreeNode<T>* fromJson(const folly::dynamic& obj) {
-  int index = obj["index"].asInt();
+TreeNode<T>* fromJson(const folly::dynamic& obj, const Config& cfg) {
+  const folly::dynamic* feature = nullptr;
+  try {
+    feature = &obj["feature"];
+  } catch(...) {
+  }
```

Comment:
check folly obj has key "feature" or "index", thus no need for try and empty catch.. 
and also keep the backward compatibility.